src/main/java/org/jabref/logic/importer/fetcher/MathSciNet.java

```diff
@@ -102,34 +115,123 @@ public URL getUrlForIdentifier(String identifier) throws URISyntaxException, Mal
-                        String bibTexFormat = entriesArray.getJSONObject(i).getString("bib");
-                        entries.addAll(bibtexParser.parseEntries(bibTexFormat));
+                        JSONObject entryObject = entriesArray.getJSONObject(i);
+                        BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                        entries.add(bibEntry);
                     }
                 } else {
                     var element = node.getObject();
-                    JSONArray entriesArray = element.getJSONObject("all").getJSONArray("results");
-                    for (int i = 0; i < entriesArray.length(); i++) {
-                        String bibTexFormat = entriesArray.getJSONObject(i).getString("bibTexFormat");
-                        entries.addAll(bibtexParser.parseEntries(bibTexFormat));
+
+                    if (element.has("all")) {
+                        JSONArray entriesArray = element.getJSONObject("all").getJSONArray("results");
+                        for (int i = 0; i < entriesArray.length(); i++) {
+                            JSONObject entryObject = entriesArray.getJSONObject(i);
+                            BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                            entries.add(bibEntry);
+                        }
+                    } else if (element.has("results")) {
+                        JSONArray entriesArray = element.getJSONArray("results");
+                        for (int i = 0; i < entriesArray.length(); i++) {
+                            JSONObject entryObject = entriesArray.getJSONObject(i);
+                            BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                            entries.add(bibEntry);
+                        }
                     }
                 }
-            } catch (JSONException | TokenMgrException e) {
+            } catch (JSONException | ParseException e) {
                 LOGGER.error("An error occurred while parsing fetched data", e);
                 throw new ParseException("Error when parsing entry", e);
             }
             return entries;
         };
     }
 
+    private BibEntry jsonItemToBibEntry(JSONObject item) throws ParseException {
+        try {
+            BibEntry entry = new BibEntry(StandardEntryType.Article);
+            // Set fields based on the mappings
+            for (Map.Entry<StandardField, List<String>> mapEntry : fieldMappings.entrySet()) {
+                StandardField field = mapEntry.getKey();
+                List<String> path = mapEntry.getValue();
+
+                String value;
+                if (field == StandardField.AUTHOR) {
+                    value = toAuthors(item.optJSONArray(path.getFirst()));
+                } else if (field == StandardField.KEYWORDS) {
+                    value = getKeywords(item.optJSONObject(path.getFirst()));
+                } else {
+                    value = getOrNull(item, path).orElse(null);
```

Comment:
Just name the method `getOther`, because it is not author or keywords

```suggestion
                    value = getOthers(item, path);
```